saleor/tests/e2e/shop/utils/preparing_shop.py

```diff
@@ -1,44 +1,168 @@
 from ...channel.utils import create_channel
-from ...shipping_zone.utils.shipping_method import create_shipping_method
-from ...shipping_zone.utils.shipping_method_channel_listing import (
+from ...shipping_zone.utils import (
+    create_shipping_method,
     create_shipping_method_channel_listing,
+    create_shipping_zone,
+)
+from ...taxes.utils import (
+    create_tax_class,
+    get_tax_configurations,
+    update_tax_class,
+    update_tax_configuration,
 )
-from ...shipping_zone.utils.shipping_zone import create_shipping_zone
 from ...warehouse.utils import create_warehouse
+from ..utils.shop_update_settings import update_shop_settings
+
+
+def create_and_update_tax_classes(e2e_staff_api_client, tax_rates):
+    created_tax_classes = {rate["type"]: {} for rate in tax_rates}
+
+    for tax_rate_info in tax_rates:
+        tax_class_data = create_tax_class(
+            e2e_staff_api_client,
+            tax_rate_info["name"],
+            [
+                {
+                    "countryCode": tax_rate_info["country_code"],
+                    "rate": tax_rate_info["rate"],
+                }
+            ],
+        )
+        created_tax_classes[tax_rate_info["type"]] = {
+            "id": tax_class_data["id"],
+            "rate": tax_rate_info["rate"],
+        }
+
+        tax_class_update_input = {
+            "updateCountryRates": [
+                {
+                    "countryCode": tax_rate_info["country_code"],
+                    "rate": tax_rate_info["rate"],
+                }
+            ]
+        }
+        update_tax_class(
+            e2e_staff_api_client, tax_class_data["id"], tax_class_update_input
+        )
+
+    return created_tax_classes
+
+
+def prepare_shop(
+    e2e_staff_api_client,
+    channels=[],
+    shop_settings=None,
+    tax_settings=None,
+):
+    if shop_settings is not None:
+        update_shop_settings(e2e_staff_api_client, input_data=shop_settings)
+
+    warehouse_id = create_warehouse(e2e_staff_api_client)["id"]
+
+    created_channels = []
+    for channel_index, channel in enumerate(channels):
+        created_channel = create_channel(
+            e2e_staff_api_client,
+            order_settings=channel["order_settings"],
+            warehouse_ids=[warehouse_id],
+        )
+        channel_id = created_channel["id"]
+        created_channels.append(
+            {
+                "id": channel_id,
+                "warehouse_id": warehouse_id,
+                "slug": created_channel["slug"],
+                "shipping_zones": [],
+                "order_settings": created_channel["orderSettings"],
+            }
+        )
+
+        for shipping_zone_index, shipping_zone in enumerate(channel["shipping_zones"]):
```

Comment:
If you will change `channel["shipping_zones"]` to `channel.get("shipping_zones", []`
And in line 95, if you will change `shipping_zone["shipping_methods"]` to `shipping_zone.get("shipping_methods", [])`
Then in tests where you don't need **ANY** shipping zones/shipping methods you will be able to call prepare_shop with:
```python
channels=[
   {"slug": "channel"}
]
```
In some tests I see below call 👇 
```python
        channels=[
             {
                 "shipping_zones": [
                     {
                         "shipping_methods": [
                             {},
                         ],
                     },
                 ],
                 "order_settings": {},
             }
         ],
```
If these tests just need a channel, then I think that you could use:
```python
channels=[
   {"slug": "channel"}
]
```

Reply:
This pattern:
```
        channels=[
             {
                 "shipping_zones": [
                     {
                         "shipping_methods": [
                             {},
                         ],
                     },
                 ],
                 "order_settings": {},
             }
         ],
```
creates default values for: channel, shipping zones, shipping methods and order settings - there is no e2e test where I need a single channel alone. Should I still implement your solution? @korycins